clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -436,6 +441,24 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
         }
     }
 
+    private static long configureDeliveryTimeout(ProducerConfig config) {
+        long deliveryTimeoutMs = config.getLong(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG);
+        long lingerMs = config.getLong(ProducerConfig.LINGER_MS_CONFIG);
+        int requestTimeoutMs = config.getInt(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG);
+
+        // If lingerMs is Long.MAX_VALUE the sum with requestTimeoutMs overflows
+        if (deliveryTimeoutMs == Long.MAX_VALUE && lingerMs > 0 && requestTimeoutMs > 0 && lingerMs + requestTimeoutMs < 0)
```

Comment:
This check is a little flaky. What if deliveryTimeoutMs is `Long.MAX_VALUE - 1`?